### PR TITLE
Rewards extension in brave bar

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#669

Adds a placeholder for Rewards extension (the real extension code is to be implemented separately).
Places the extension into resource pack and provides necessary wiring to load the extension from resources. The loading of extension is predicated on BRAVE_REWARDS_ENABLED flag due to pending port of anonize.
Removes the Rewards extension button from main extensions area (ToolbarActionsBar / BrowserActionsContainer) and places it into BraveActionsBar in the omnibox.
Rewards extension button inherits behavior of BraveActionsBar buttons: disabled context menu, no drag&drop of buttons within the bar.
Renames ShieldsActionViewController to BraveActionViewController as the controller is now used for both Shields and Rewards action views. Due to renaming patches/chrome-browser-ui-extensions-extension_action_view_controller.h.patch had to be updated with the new name.

This PR should not be merged because it contains placeholder UI for Rewards extension.

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Needed or QA/No-QA-Needed) to include the closed issue in milestone 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by mkarolin** [comment]
@petemill, thanks, I was going to ping Ross about the icons. If you are going to get them, should we add all sizes like we do for the shields extension (vendor/brave-extension/app/assets/img)? Please, let me know if you want me to follow up.

File: browser/resources/resource_ids [hunk 1]
```diff
@@ -22,4 +22,7 @@
     "includes": [36500],
   },
   # brave webtorrent 37500
+  "brave/components/brave_rewards/resources.grd": {
+    "includes": [37600],
```
Comment (yrliou) [inline browser/resources/resource_ids]: As discussed offline, let's do 38000 here, thanks!

File: browser/ui/toolbar/brave_toolbar_actions_model.cc [hunk 2]
```diff
@@ -9,9 +9,10 @@
 #include "extensions/common/extension.h"
 
 bool BraveToolbarActionsModel::ShouldAddExtension(const extensions::Extension* extension) {
-  // Don't show the Brave 'extension' in the ToolbarActions extensions area. It
-  // will instead be shown in the BraveActions area.
-  if (extension->id() == brave_extension_id) {
+  // Don't show the Brave and Rewards 'extensions' in the ToolbarActions
+  // extensions area. They will instead be shown in the BraveActions area.
+  if (extension->id() == brave_extension_id || 
+	  extension->id() == brave_rewards_extension_id) {
```
Comment (yrliou) [inline browser/ui/toolbar/brave_toolbar_actions_model.cc]: nit: align with previous line?

File: components/brave_rewards/extension/manifest.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "version": "1.0.0",
+  "name": "Brave Rewards",
+  "manifest_version": 2,
+  "description": "Brave Rewards Extension",
+  "browser_action": {
+    "default_popup": "extension/popup.html",
+    "default_icon": "extension/img/BAT_icon_placeholder_32.png"
+  },
```
Comment (yrliou) [inline components/brave_rewards/extension/manifest.json]: extensions should have CSP defined, tho probably fine since it's still just a stub right now.

Comment (mkarolin) [reply components/brave_rewards/extension/manifest.json]: Right, my assumption is that whomever implements the actual extension will add the appropriate CSP.

**Comment by petemill** [comment]
@mkarolin here's some updated icons for the extension if we want to get that in here before it's merged https://www.dropbox.com/s/370cc0dtowvh45p/brave-payments-extension-icon.zip?dl=0

Additionally, it would be great to squash some of these commits so that anything fixed or any changes in response to reviews are squashed in to the commit that the change originated in.

**Comment by mkarolin** [comment]
Updated with actual extension icons. Squashed code review commits.

**Comment by petemill** [comment]
0.55.x 45cc8b36

**Comment by bbondy** [comment]
added merged/0.55.x label

**Pull request merged**